File: CS/DXExample.Module/SelectFirstNavigationItemControllerBase.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp.SystemModule;

namespace DXExample.Module {
    public abstract class SelectFirstNavigationItemControllerBase : WindowController {
        public const string AllowSelectFirstItemAttributeName = "AllowSelectFirstItem";
        SingleChoiceAction navigationAction;
        public SelectFirstNavigationItemControllerBase() {
            TargetWindowType = WindowType.Main;
        }
        protected override void OnActivated() {
            base.OnActivated();
            navigationAction = Frame.GetController<ShowNavigationItemController>().ShowNavigationItemAction;
            Frame.TemplateChanged += new EventHandler(Frame_TemplateChanged);
        }
        protected override void OnDeactivating() {
            base.OnDeactivating();
            Frame.TemplateChanged -= new EventHandler(Frame_TemplateChanged);
        }
        private void Frame_TemplateChanged(object sender, EventArgs e) {
            IActionContainer navigationControl = FindNavBarControl();
            if (navigationControl != null) {
                SetupNavigationControl(navigationControl);
            }
        }
        protected abstract IActionContainer FindNavBarControl();
        protected abstract void SetupNavigationControl(IActionContainer navigationControl);
        protected void SelectFirstItem(int navGroupIndex) {
            SelectFirstItem(navigationAction.Items[navGroupIndex]);
        }
        protected void SelectFirstItem(ChoiceActionItem navGroup) {
            if (!AllowSelectFirstItem()) return;
            foreach (ChoiceActionItem item in navGroup.Items) {
                if (item.Enabled.ResultValue && item.Active.ResultValue) {
                    navigationAction.DoExecute(item);
                    return;
                }
            }
        }
        protected bool AllowSelectFirstItem() {
            DictionaryAttribute attr = Application.Model.RootNode.FindChildNode(ShowNavigationItemController.NavigationItemsNodeName).FindAttribute(AllowSelectFirstItemAttributeName);
            if (attr != null && !string.IsNullOrEmpty(attr.Value)) {
                return Convert.ToBoolean(attr.Value);
            }
            return false;
        }
        public override Schema GetSchema() {
            return new Schema(new DictionaryXmlReader().ReadFromString(
                @"<Element Name=""Application"">
                    <Element Name=""" + ShowNavigationItemController.NavigationItemsNodeName + @""">
                       <Attribute Name=""" + AllowSelectFirstItemAttributeName + @""" Choice=""True,False""/>
                    </Element>
                </Element>"
                )
            );
        }
    }
}

File: VB/DXExample.Module/SelectFirstNavigationItemControllerBase.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp.SystemModule

Namespace DXExample.Module
	Public MustInherit Class SelectFirstNavigationItemControllerBase
		Inherits WindowController
		Private showNavigationItemController As ShowNavigationItemController
		Protected navigationAction As SingleChoiceAction
		Public Sub New()
			TargetWindowType = WindowType.Main
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			showNavigationItemController= Frame.GetController(Of ShowNavigationItemController)()
			navigationAction = showNavigationItemController.ShowNavigationItemAction
			AddHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
		End Sub
		Protected Overrides Sub OnDeactivated()
			MyBase.OnDeactivated()
			RemoveHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
		End Sub
		Private Sub Frame_TemplateChanged(ByVal sender As Object, ByVal e As EventArgs)
			Dim navigationControl As IActionContainer = FindNavigationActionContainer()
			If navigationControl IsNot Nothing Then
				SetupNavigationControl(navigationControl)
			End If
		End Sub
		Protected MustOverride Function FindNavigationActionContainer() As IActionContainer
		Protected MustOverride Sub SetupNavigationControl(ByVal container As IActionContainer)
		'protected void SelectFirstItem(int navGroupIndex) {
		'    AutoSelectFirstItemInGroup(navigationAction.Items[navGroupIndex]);
		'}
		Protected Sub AutoSelectFirstItemInGroup(ByVal navGroupItem As ChoiceActionItem)
			If (Not CanAutoSelectFirstItemInGroup()) Then
				Return
			End If
			For Each item As ChoiceActionItem In navGroupItem.Items
				If item.Enabled.ResultValue AndAlso item.Active.ResultValue Then
					navigationAction.DoExecute(item)
					Return
				End If
			Next item
		End Sub
		Protected Function CanAutoSelectFirstItemInGroup() As Boolean
			Return (CType((CType(Application.Model, IModelApplicationNavigationItems)).NavigationItems, IModelRootNavigationItemsEx)).AllowSelectFirstItem
		End Function
	End Class
End Namespace
